tests/acceptance/blocking_adapter_test.py

```diff
@@ -476,13 +469,13 @@ def test(self):
         self.assertIsInstance(method, pika.spec.Basic.GetOk)
         self.assertEqual(method.delivery_tag, 1)
         self.assertFalse(method.redelivered)
-        self.assertEqual(method.exchange, _p3_as_bytes(''))
-        self.assertEqual(method.routing_key, q_name)
+        self.assertEqual(method.exchange, as_bytes(''))
+        self.assertEqual(method.routing_key, as_bytes(q_name))
```

Comment:
Here there is no visible asymmetry: method is a protocol class instance and never available for users (as much as anything can be made unavailable in python). This test asserts that the implementation detail of method containing attributes with certain values holds.

I confess not being well-versed in AMQP either, but: Pika did unicode to utf-8 bytes conversions in pika/spec.py before we started with the py3 changes. This is why ``as_bytes`` does make sense here.

Please have a look at how utils/codegen.py generates "string handling" for protocol methods. It was and still is unicode aware.